core/src/main/java/org/mule/runtime/core/api/policy/PolicyChain.java

```diff
@@ -53,7 +83,31 @@ public void setProcessors(List<Processor> processors) {
 
   @Override
   public final void initialise() throws InitialisationException {
-    processorChain = new DefaultMessageProcessorChainBuilder().chain(this.processors).build();
+    DefaultMessageProcessorChainBuilder chainBuilder = new DefaultMessageProcessorChainBuilder().chain(processors);
+
+    List<Processor> beforeNext = new ArrayList<>();
+    List<Processor> afterNext = new ArrayList<>();
+
+    boolean seenNext = false;
+    for (Processor processor : processors) {
+      if (processor instanceof PolicyNextActionMessageProcessor) {
+        seenNext = true;
+        continue;
+      }
+
+      if (seenNext) {
+        afterNext.add(processor);
+      } else {
+        beforeNext.add(processor);
+      }
+    }
+
+    processingStrategy = new SourcePolicyProcessingStrategy(schedulerService, muleContext.getSchedulerBaseConfig(),
```

Comment:
Does this mean that each and every policy will have its own processing strategy and set of pools? Isn't there a way which can define a single processingStrategy and then reuse it for all policies?

Reply:
the pool is the same in the end. Only the scheduler is created per policy.